src/config.c

```diff
@@ -65,6 +66,24 @@ void reset_cloud_disconnect_reason(ParodusCfg *cfg)
 	cfg->cloud_disconnect = NULL;
 }
 
+void set_cloud_status(char *status)
+{
+    pthread_mutex_lock(&config_mut);	
+    get_parodus_cfg()->cloud_status = strdup(status);
```

Comment:
Add a null check for status

Reply:
Resolved in latest change , this config change is vasuki's changes FYI